src/main/java/io/leonis/subra/ipc/network/GameDeducer.java

```diff
@@ -2,42 +2,40 @@
 
 import io.leonis.algieba.Temporal;
 import io.leonis.subra.game.data.*;
-import io.leonis.subra.ipc.network.GamePublisher.GameFrame;
+import io.leonis.subra.ipc.network.GameDeducer.GameFrame;
 import io.leonis.subra.ipc.serialization.protobuf.*;
 import io.leonis.subra.ipc.serialization.protobuf.SSLVisionDeducer.VisionPacket;
 import io.leonis.subra.ipc.serialization.protobuf.vision.GoalsDeducer;
 import io.leonis.zosma.game.engine.*;
 import java.util.Set;
 import lombok.*;
 import lombok.experimental.Delegate;
-import org.reactivestreams.*;
+import org.reactivestreams.Publisher;
 import org.robocup.ssl.Referee.SSL_Referee;
 import org.robocup.ssl.Wrapper.WrapperPacket;
 import reactor.core.publisher.Flux;
 
 /**
- * The Class GamePublisher.
+ * The Class GameDeducer.
  *
- * This class contains the functionality of a {@link Publisher} of {@link GameFrame}.
+ * This class deduces a {@link WrapperPacket} and {@link SSL_Referee} to a {@link GameFrame}.
  *
  * @author Rimon Oz
  */
 @AllArgsConstructor
-public final class GamePublisher implements Publisher<GameFrame> {
-  private final Publisher<WrapperPacket> visionPublisher;
-  private final Publisher<SSL_Referee> refboxPublisher;
+public final class GameDeducer<I extends WrapperPacketSupplier & SSLRefereeSupplier> implements Deducer<I, GameFrame> {
 
   @Override
-  public void subscribe(final Subscriber<? super GameFrame> subscriber) {
-    Flux.combineLatest(
-        Flux.from(this.visionPublisher).transform(new SSLVisionDeducer()),
-        Flux.from(this.refboxPublisher).transform(new SSLRefboxDeducer()),
-        GameFrameWithoutGoals::new)
-      .transform(new ParallelDeducer<>(
+  public Publisher<GameFrame> apply(final Publisher<I> entryPublisher) {
+    return Flux.from(entryPublisher)
+        .transform(new ParallelDeducer<>(
+            new SSLVisionDeducer<>(),
+            new SSLRefboxDeducer<>(),
+            GameFrameWithoutGoals::new))
```

Comment:
on r51 you want to have 

```
    <J extends Player.SetSupplier & GoalDimension.Supplier & Field.Supplier & Ball.SetSupplier>
    GameFrameWithoutGoals(final J packet, final Referee.Supplier referee) {
```

in order to get rid of the explicit dependency on `VisionPacket`